### PR TITLE
Docker!

### PR DESCRIPTION
**Pull request by chilloutman**
With this I can launch our robopatrol ROS node inside a Docker container!

The turtlebot.launch file is not working yet. I might need some help with that one. @sidloki?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM robopatrol:robopatrol-base
```
Comment (DeeVeX) [inline Dockerfile]: `FROM robopatrol/robopatrol-base:latest`

Otherwise the container can not be found and pulled.

**Comment by chilloutman** [comment]
Current issue is that the turtlebot container does not want to launch:

```
robot         | [rosrun] Couldn't find executable named turtlebot below /catkin_ws/src/robopatrol
```

What we execute is:

``` shell
source ./devel/setup.bash && rosrun robopatrol turtlebot
```

**Comment by chilloutman** [comment]
Just got further by using `roslaunch` instead of `rosrun`:

```
 > docker-compose up robot                                                                                  . (docker*)
Recreating robot
Attaching to robot
robot         | Traceback (most recent call last):
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/__init__.py", line 307, in main
robot         |     p.start()
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/parent.py", line 268, in start
robot         |     self._start_infrastructure()
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/parent.py", line 217, in _start_infrastructure
robot         |     self._load_config()
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/parent.py", line 132, in _load_config
robot         |     roslaunch_strs=self.roslaunch_strs, verbose=self.verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/config.py", line 451, in load_config_default
robot         |     loader.load(f, config, verbose=verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 730, in load
robot         |     self._load_launch(launch, ros_config, is_core=core, filename=filename, argv=argv, verbose=verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 702, in _load_launch
robot         |     self._recurse_load(ros_config, launch.childNodes, self.root_context, None, is_core, verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 666, in _recurse_load
robot         |     val = self._include_tag(tag, context, ros_config, default_machine, is_core, verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 95, in call
robot         |     return f(*args, **kwds)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 609, in _include_tag
robot         |     default_machine, is_core, verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 666, in _recurse_load
robot         |     val = self._include_tag(tag, context, ros_config, default_machine, is_core, verbose)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 95, in call
robot         |     return f(*args, **kwds)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 587, in _include_tag
robot         |     inc_filename = self.resolve_args(tag.attributes['file'].value, context)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/xmlloader.py", line 183, in resolve_args
robot         |     return substitution_args.resolve_args(args, context=context.resolve_dict, resolve_anon=self.resolve_anon)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 312, in resolve_args
robot         |     resolved = _resolve_args(resolved, context, resolve_anon, commands)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 325, in _resolve_args
robot         |     resolved = commands[command](resolved, a, args, context)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 141, in _find
robot         |     source_path_to_packages=source_path_to_packages)
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/roslaunch/substitution_args.py", line 184, in _find_executable
robot         |     full_path = _get_executable_path(rp.get_path(args[0]), path)
robot         |   File "/usr/lib/python2.7/dist-packages/rospkg/rospack.py", line 200, in get_path
robot         |     raise ResourceNotFound(name, ros_paths=self._ros_paths)
robot         | ResourceNotFound: rosbridge_server
robot         | ROS path [0]=/opt/ros/indigo/share/ros
robot         | ROS path [1]=/catkin_ws/src
robot         | ROS path [2]=/opt/ros/indigo/share
robot         | ROS path [3]=/opt/ros/indigo/stacks
robot         | ... logging to /root/.ros/log/f2e6f8b2-0b73-11e6-832f-0242ac130002/roslaunch-33da51967baf-67.log
robot         | Checking log directory for disk usage. This may take awhile.
robot         | Press Ctrl-C to interrupt
robot         | Done checking log file disk usage. Usage is <1GB.
robot         | 
robot exited with code 1
```

**Comment by chilloutman** [comment]
I think we're pretty close now!

Whats missing now:
- `EXPOSE` correct ports for ROS bridge and possibly other stuff. (Add to `Dockerfile` and `docker-compose.yml`)
- Connect to ROS bridge and see what breaks next. :-p

Because I'm such a ROS-noob, It's difficult for me to verify if it's working. Some help would be appreciated!

**Comment by chilloutman** [comment]
@sidloki, which step did I miss?

``` sh
robot         | Traceback (most recent call last):
robot         |   File "/opt/ros/indigo/lib/rosbridge_server/rosbridge_websocket", line 43, in <module>
robot         |     from rosbridge_server import RosbridgeWebSocket
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/rosbridge_server/__init__.py", line 3, in <module>
robot         |     from udp_handler import RosbridgeUdpSocket,RosbridgeUdpFactory
robot         |   File "/opt/ros/indigo/lib/python2.7/dist-packages/rosbridge_server/udp_handler.py", line 6, in <module>
robot         |     from twisted.internet.protocol import DatagramProtocol,Factory
robot         | ImportError: No module named twisted.internet.protocol
```

**Comment by chilloutman** [comment]
`apt-get install python-twisted` did the trick!

You can try it now!  `docker-compose up`

**Comment by chilloutman** [comment]
`docker-machine up simulation` to start a simulation.

`docker-machine up turtlebot` to run on the real turtlebot.

**Pull request merged**